apps/exam/views.py

```diff
@@ -55,13 +55,12 @@ def post(self, request, exam_id):
     Pre-populate attempt_questions table with questions from this exam.
     """
     exam = Exam.objects.get(id=exam_id)
-    exam_duration_minutes = exam.duration_minutes
-
-    if request.user.requires_extra_time:
-      exam_duration_minutes += 30
 
     current_attempt = ExamAttempt.objects.create(
-        user=request.user, exam=exam, duration_minutes=exam_duration_minutes)
+        user=request.user,
+        exam=exam,
+        duration_minutes=exam.duration_minutes +
```

Comment:
I suggest using ExamAttempt save() method as attempt already has access to user.